### PR TITLE
Prevent writing invalid mtime

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ <skjnldsv@protonmail.com>

### CODE REVIEW

**Review by mgallien — COMMENTED** [review]
but is it really enough to prevent files from being stored with old invalid mtime ?
is it possible to instead return an error to the client ?

**Comment by skjnldsv** [comment]
> but is it really enough to prevent files from being stored with old invalid mtime ? is it possible to instead return an error to the client ?

done

**Review by nickvergessen — APPROVED** [review]

File: apps/dav/lib/Connector/Sabre/Node.php [hunk 1]
```diff
@@ -412,6 +412,11 @@ protected function sanitizeMtime($mtimeFromRequest) {
 			throw new \InvalidArgumentException('X-OC-MTime header must be an integer (unix timestamp).');
 		}
 
+		// Prevent writing invalid mtime (timezone-proof)
+		if ((int)$mtimeFromRequest <= 24 * 60 * 60) {
+			throw new \InvalidArgumentException('X-OC-MTime header must be a valid positive integer');
+		}
```
Comment (come-nc) [inline apps/dav/lib/Connector/Sabre/Node.php:418]: PHP5 is not supported anymore so the hexadecimal check is obsolete.
Might as well merge the if as one:
`if (!is_numeric($mtimeFromRequest) || ((int)$mtimeFromRequest <= 24 * 60 * 60)) {`

Comment (nickvergessen) [reply apps/dav/lib/Connector/Sabre/Node.php:418]: I think it's good to have 2 different errors for a non-numeric and a invalid numeric value

**Comment by skjnldsv** [comment]
Tests fixed

**Review by come-nc — APPROVED** [review]

**Comment by mgallien** [comment]
@skjnldsv 
as files got truncated on Monday after a version of this got applied I decided to test against it and a desktop client sending invalid modified time
the fact is that with what was merged in stable23 I got a 500 internal server error reported and the files is still written on storage

**Pull request merged**